sys/include/event/periodic.h

```diff
@@ -76,13 +82,27 @@ void event_periodic_init(event_periodic_t *event_periodic, ztimer_clock_t *clock
  * @param[in]   event_periodic   event_timout context object to use
  * @param[in]   interval         period length for the event
  */
-static inline void event_periodic_start(event_periodic_t *event_periodic,
-                                        uint32_t interval)
+static inline void event_periodic_start(event_periodic_t *event_periodic, uint32_t interval)
 {
     event_periodic->timer.interval = interval;
     ztimer_periodic_start(&event_periodic->timer);
 }
 
+/**
+ * @brief   Set the amount of times the periodic event should repeat itself.
+ *
+ * @pre     Must be called before @ref event_periodic_start
+ *
+ * @param[in]   event_periodic   event_timout context object to use
+ * @param[in]   count            times the event should repeat itself,
+ *                               EVENT_PERIODIC_FOREVER to run for ever.
+ */
+static inline void event_periodic_set_count(event_periodic_t *event_periodic,
+                                            uint32_t count)
+{
+    event_periodic->count = count;
```

Comment:
Either this must only be called on a stopped timer or there is a race condition here

Reply:
Yes, I added a pre-condition, but maybe I should add an assert?

Reply:
Or maybe wrap it in `irq_disable()` / `irq_restore()`